shell/platform/darwin/ios/framework/Source/FlutterObservatoryPublisher.mm

```diff
@@ -115,7 +112,7 @@ - (void)publishServiceProtocolPort:(std::string)uri {
   if (err != 0) {
     FML_LOG(ERROR) << "Failed to register observatory port with mDNS.";
   } else {
-    DNSServiceProcessResult(_dnsServiceRef);
+    DNSServiceSetDispatchQueue(_dnsServiceRef, [NSOperationQueue currentQueue].underlyingQueue);
```

Comment:
[The Apple docs](https://developer.apple.com/documentation/foundation/nsoperationqueue/1415344-underlyingqueue?language=objc) say that the default value for `underlyingQueue` is `nil`.  Is that something we need to worry about here?

Reply:
I believe we're ok because we're taking this from an NSOperationQueue that is clearly running (it's the one we're on), but I'd defer to @chinmaygarde or @cbracken  on that one.

Reply:
FWIW, this is working consistently when I test it on a simulator.  And our worst case scenario here is that the mDNS publishing would fail and we wouldn't get notified.

Reply:
Personally I'd prefer we were explicit about the queue we put these on, for readability's sake. e.g. `dispatch_get_main_queue()` for the platform thread.

Reply:
I'll change it to that - I was under the impression that was deprecated, but I think I was mixing it up with something else now.